qiita_ware/commands.py

```diff
@@ -66,34 +60,42 @@ def submit_EBI(preprocessed_data_id, action, send, fastq_dir_fp=None):
         # step 4: sending sequences
         old_ascp_pass = environ.get('ASPERA_SCP_PASS', '')
         environ['ASPERA_SCP_PASS'] = qiita_config.ebi_seq_xfer_pass
-        seqs_cmds = ebi_submission.generate_send_sequences_cmd()
+
         LogEntry.create('Runtime',
                         ("Submitting sequences for pre_processed_id: "
                          "%d" % preprocessed_data_id))
-        try:
-            # place holder for moi call see #1477
-            pass
-        except:
-            LogEntry.create('Fatal', seqs_cmds,
-                            info={'ebi_submission': preprocessed_data_id})
-        else:
-            LogEntry.create('Runtime',
-                            ('Submission of sequences of pre_processed_id: '
-                             '%d completed successfully' %
-                             preprocessed_data_id))
+        for cmd in ebi_submission.generate_send_sequences_cmd():
+            cmd_pieces = shsplit(cmd)
+            try:
+                call(cmd_pieces, stdout=open(ebi_submission.ascp_reply, 'a'))
+            except:
+                with open(ebi_submission.ascp_reply, 'r') as f:
+                    content = f.read()
+                ebi_submission.study.ebi_submission_status = (
+                    "failed: ASCP - %s" % content)
+                LogEntry.create('Fatal', content,
```

Comment:
Cool, just wanted to avoid cluttering the logging page via the web
interface.

On (Oct-09-15|10:18), Antonio Gonzalez wrote:
> > -                            info={'ebi_submission': preprocessed_data_id})
> > -        else:
> > -            LogEntry.create('Runtime',
> > -                            ('Submission of sequences of pre_processed_id: '
> > -                             '%d completed successfully' %
> > -                             preprocessed_data_id))
> > +        for cmd in ebi_submission.generate_send_sequences_cmd():
> > +            cmd_pieces = shsplit(cmd)
> > +            try:
> > +                call(cmd_pieces, stdout=open(ebi_submission.ascp_reply, 'a'))
> > +            except:
> > +                with open(ebi_submission.ascp_reply, 'r') as f:
> > +                    content = f.read()
> > +                ebi_submission.study.ebi_submission_status = (
> > +                    "failed: ASCP - %s" % content)
> > +                LogEntry.create('Fatal', content,
> 
> The only time I have seen this fail is when the login is incorrect and the reply is something like: "Wong login" or something like that. So, no?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/qiita/pull/1488/files#r41654817